### DISCUSSION TITLE
Berry - Switch/Button State

### DISCUSSION THREAD

**Discussion by ChrisHolter** [arendst/Tasmota#15717] [Q&A] [answered]
Hi,

is there a way in Berry to get the State of Switch/Button independent of an relay?

**Comment by s-hadinger** [comment, marked as answer]
The best way is to create Berry rules to capture events from switches and buttons. 

Alternatively you can read low level gpio states with `gpio.digital_read` but doing such you bypass all the logic applied by Tasmota like debouncing

**Reply by ChrisHolter** [reply]
Thx, Using rules did the job!

**Reply by sfromis** [reply]
You also have the call `tasmota.get_switches()` returning a list of booleans, one for each switch. Not available for Buttons, which basically are intended for momentary push buttons, where it works better to capture events.

**Reply by ReinhardDaemon** [reply]
Is there no way like:

switch_status = tasmota.get_switches(1) 

to get the boolean value of switch1?

**Reply by ReinhardDaemon** [reply]
> You also have the call `tasmota.get_switches()` returning a list of booleans, one for each switch. Not available for Buttons, which basically are intended for momentary push buttons, where it works better to capture events.

**How to get the information (false, true) out of tasmota.get_switches(), so that I can go on working in the Berry script?**
is it like:

if (tasmota.get_switches()[0]) == TRUE) .... ?

**Reply by sfromis** [reply]
If what you want is a state, configuring it as a Button is not the right thing, as a Button is for a momentary toggle action, not for having it in a certain state. Hence you should configure it as a Switch, and as the array you get from `tasmota.get_switches` already has boolean values, you can simply do like this:
`if tasmota.get_switches()[0] print('Switch1 is set') end`
Comparing a boolean to true makes no difference. Beware that normal wiring (ground and pull-up) means true when not pushed, and false when pushed.
